pkgs/native_assets_builder/test_data/treeshaking_native_assets/hook/build.dart

```diff
@@ -32,24 +33,27 @@ void main(List<String> arguments) async {
         }),
       linkInPackage: packageName,
     );
-    _placeholderAsset(config, output);
+    output.addAsset(
+      placeholderAsset(config),
+      linkInPackage: packageName,
+    );
   });
 }
 
 // TODO: Currently, linking cannot add assets, so we add a placeholder here.
 // See also https://github.com/dart-lang/native/issues/1084.
-void _placeholderAsset(BuildConfig config, BuildOutput output) {
-  final uri = config.outputDirectory.resolve(
-      config.targetOS.libraryFileName(packageName, DynamicLoadingBundled()));
-  File.fromUri(uri).createSync(recursive: true);
-  output.addAsset(
-    NativeCodeAsset(
-      package: packageName,
-      name: 'src/${packageName}_bindings.dart',
-      linkMode: DynamicLoadingBundled(),
-      file: uri,
-      os: config.targetOS,
-    ),
-    linkInPackage: packageName,
+NativeCodeAsset placeholderAsset(HookConfig config) {
```

Comment:
This should not be needed anymore.

Am I correct in thinking that:
1. The unit tests for the package here can land without SDK changes.
2. Then we land the SDK CL in which this test_data is used in dartdev unit tests and we actually add a new asset in link.
3. Then we land another PR in the dart-lang/native that enables the examples.

If I am not mistaken, we should not need to land the placeholder workaround.